Gems/GeoJSONSpawner/Code/Source/GeoJSONSpawner/GeoJSONSpawnerComponent.cpp

```diff
@@ -0,0 +1,262 @@
+                Spawn(m_cachedGeoJson);
+            }
+        }
+
+        if (m_isInModifyProcess)
+        {
+            if (m_ticketsToDespawn == 0)
+            {
+                m_isInModifyProcess = false;
+                SpawnModifiedEntities(m_cachedGeoJson);
+            }
+        }
+    }
+
+    void GeoJSONSpawnerComponent::SpawnEntities()
+    {
+        m_spawnableTickets.clear();
+        const auto geometryObjectInfo = GeoJSONUtils::ParseJSONFromFile(m_geoJsonFilePath.c_str());
+        m_spawnableEntityInfo =
+            GeoJSONUtils::GetSpawnableEntitiesFromGeometryObjectVector(geometryObjectInfo, m_spawnableAssetConfigurations);
+        m_spawnableTickets = GeoJSONUtils::SpawnEntities(
+            m_spawnableEntityInfo, m_spawnableAssetConfigurations, m_defaultSeed, AzPhysics::DefaultPhysicsSceneName, GetEntityId());
+    }
+
+    void GeoJSONSpawnerComponent::Spawn(const AZStd::string& rawJsonString)
+    {
+        if (!m_spawnableTickets.empty() && !m_isInSpawningProcess)
+        {
+            m_isInSpawningProcess = true;
+            m_cachedGeoJson = rawJsonString;
+            DespawnAllEntities();
+        }
+        else
+        {
+            const auto result = GeoJSONUtils::ParseJSONFromRawString(rawJsonString);
+            m_spawnableEntityInfo = GeoJSONUtils::GetSpawnableEntitiesFromGeometryObjectVector(result, m_spawnableAssetConfigurations);
+
+            m_spawnableTickets = GeoJSONUtils::SpawnEntities(
+                m_spawnableEntityInfo, m_spawnableAssetConfigurations, m_defaultSeed, AzPhysics::DefaultPhysicsSceneName, GetEntityId());
+        }
+    }
+
+    void GeoJSONSpawnerComponent::SpawnModifiedEntities(const AZStd::string& rawJsonString)
+    {
+        const auto result = GeoJSONUtils::ParseJSONFromRawString(rawJsonString);
+        auto entityInfos = GeoJSONUtils::GetSpawnableEntitiesFromGeometryObjectVector(result, m_spawnableAssetConfigurations);
+
+        const auto spawnableTickets = GeoJSONUtils::SpawnEntities(
+            entityInfos, m_spawnableAssetConfigurations, m_defaultSeed, AzPhysics::DefaultPhysicsSceneName, GetEntityId());
+
+        for (const auto& pair : spawnableTickets)
+        {
+            m_spawnableTickets[pair.first] = AZStd::move(pair.second);
+        }
+    }
+
+    void GeoJSONSpawnerComponent::Modify(const AZStd::string& rawJsonString)
+    {
+        const auto groupIdsToModify = GeoJSONUtils::ExtractIdsFromRawString(rawJsonString);
+
+        for (const auto& groupId : groupIdsToModify)
+        {
+            if (m_spawnableTickets.find(groupId) == m_spawnableTickets.end())
+            {
+                AZ_Error(
+                    "GeoJSONSpawnerComponent",
+                    false,
+                    "Cannot modify entities with ID: %d. Such group does not exist. Action aborted.",
+                    groupId);
+                return;
+            }
+        }
+
+        m_isInModifyProcess = true;
```

Comment:
I propose to push a functor to a queue in tickbus instead of using booleans.

Reply:
This a place where I would call AZ::TickBus::Handler::Connect()!